monitor/data_monitoring.py

```diff
@@ -76,74 +71,80 @@ def _split_list_to_chunks(items: list, chunk_size: int = 50) -> List[List]:
             chunk_list.append(items[i: i + chunk_size])
         return chunk_list
 
-    def _update_sent_alerts(self, alert_ids) -> None:
+    def _update_sent_alerts(self, alert_ids: List[str], table_name: str) -> None:
         alert_ids_chunks = self._split_list_to_chunks(alert_ids)
         for alert_ids_chunk in alert_ids_chunks:
-            self.dbt_runner.run_operation(macro_name='update_sent_alerts',
-                                          macro_args={'alert_ids': alert_ids_chunk},
-                                          json_logs=False)
-
-    def _query_alerts(self, days_back: int) -> AlertsQueryResult:
-        results = self.dbt_runner.run_operation(macro_name='get_new_alerts', macro_args={'days_back': days_back})
-        test_result_alerts = []
-        failed_to_parse_alert_dicts = []
-        if results:
-            test_result_alert_dicts = json.loads(results[0])
-            self.execution_properties['alert_rows'] = len(test_result_alert_dicts)
-            for test_result_alert_dict in test_result_alert_dicts:
-                test_result = TestResult.create_test_result_from_dict(test_result_alert_dict)
-                if test_result:
-                    test_result_alerts.append(test_result)
-                else:
-                    failed_to_parse_alert_dicts.append(test_result_alert_dict)
-                    self.success = False
-
-        return AlertsQueryResult(test_result_alerts, failed_to_parse_alert_dicts)
-
-    def _send_alerts_to_slack(self, alerts: List[SlackAlert]) -> None:
-        sent_alerts = []
+            self.dbt_runner.run_operation(
+                macro_name='update_sent_alerts',
+                macro_args={'alert_ids': alert_ids_chunk, 'table_name': table_name},
+                json_logs=False
+            )
+
+    def _query_alerts(self, days_back: int) -> Alerts:
```

Comment:
WDYT about making it a bit more structured and organized?
After splitting the data monitoring into Alerts and Report, we can have a BaseAlerts class that has dbt_runner, _query_alerts method, _query_alert_type method.
Then we can create TestAlerts, ModelAlerts, etc that inheritance from the BaseAlerts method.
That way each class is responsible for its own type of alerts, and is separated from the rest of the alerts types (will help disabling/enabling alerts by type + creating new alerts types).
Then we will have a workflow that calls the relevant alerts.

Reply:
Interesting. Could you please elaborate on the difference between what you're talking about and the current alerts model? As far as I see, `Alert` is your `BaseAlert` and there's `TestAlert` and `ModelAlert` that inherit from it.

Reply:
So currently Alert is more of a schema/class that represents a single alert data with formatting options, while BaseAlerts is a base class for handling multiple alerts of the same type like Test, Model, etc.
It shares the base logic for handling alerts like querying it, parsing it, and handling broken alerts that couldn't be formatted.